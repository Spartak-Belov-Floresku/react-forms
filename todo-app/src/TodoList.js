import { useState  } from "react";
import { v4 as uuid } from 'uuid';

import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
   
    const[todos, setToDos] = useState([]);

    const addToDo = newToDo => setToDos(todos => [...todos, {...newToDo, id: uuid()}]);

    const removeToDo = divId => setToDos(todos.filter(div => div.id !== divId));

    return (
        <>
        <NewTodoForm addToDo={addToDo} />
        {todos.lenght !== 0? todos.map(({id, task}) => <Todo id={id} task={task} key={id} removeToDo={removeToDo} />):''}
        </>
    )
}

export default TodoList;