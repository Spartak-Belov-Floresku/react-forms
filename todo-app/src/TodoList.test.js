import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", () => {

    render(<TodoList />);

});

it("matches snapshot", () => {

    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();

});

if("should add new todo and remove todo", () => {

    const { queryByText, getByLabelText } = render(<TodoList />);
    const input = getByLabelText("Add new task: ");
    const addToDo = queryByText("Add Task");

    expect(queryByText('The next task to do')).not.toBeInTheDocument();
    fireEvent.change(input, { target: { value: 'The next task to do' } });
    fireEvent.click(addToDo);
    expect(queryByText('The next task to do')).toBeInTheDocument();

    const removeToDo = queryByText('X');
    fireEvent.click(removeToDo);
    expect(queryByText('The next task to do')).not.toBeInTheDocument();

});