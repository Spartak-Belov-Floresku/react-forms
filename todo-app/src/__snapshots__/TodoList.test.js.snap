// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Add new task: 
    </label>
    <input
      id="task"
      name="task"
      placeholder="New task to do"
      type="text"
      value=""
    />
    <button>
      Add Task
    </button>
  </form>
</DocumentFragment>
`;
